### PR TITLE
Improve broadcast messages

### PR DESCRIPTION
**Pull request by vyneer**
Makes the broadcast messages look like other events, and shows who actually sent the broadcast out (unless it's a server message, of course).

image

### CODE REVIEW

File: assets/chat/js/user.js [hunk 1]
```diff
@@ -53,12 +53,12 @@ class ChatUser {
...
-      this.displayName = user.nick || '';
+      this.id = user?.id || null;
+      this.displayName = user?.nick || '';
       this.username = this.displayName.toLowerCase();
-      this.createdDate = user.createdDate || '';
-      this.features = user.features || [];
-      this.watching = user.watching || null;
+      this.createdDate = user?.createdDate || '';
+      this.features = user?.features || [];
+      this.watching = user?.watching || null;
```
Comment (11k) [inline assets/chat/js/user.js:61]: Do we still need to be nullsafe here considering "userless" broadcasts are now sent by the system user instead?

File: assets/chat/js/chat.js [hunk 2]
```diff
@@ -1257,12 +1258,19 @@ class Chat {
       if (!this.backlogloading) {
         const retryMilli = Math.floor(Math.random() * 30000) + 4000;
         setTimeout(() => window.location.reload(true), retryMilli);
+
         MessageBuilder.broadcast(
-          `Restart incoming in ${Math.round(retryMilli / 1000)} seconds ...`,
+          `Restart incoming in ${Math.round(retryMilli / 1000)} seconds...`,
+          new ChatUser(),
```
Comment (11k) [inline assets/chat/js/chat.js:1264]: Would you mind passing in an object with the system user's properties to the constructor here?

File: assets/chat/js/messages/ChatBroadcastMessage.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+export default class ChatBroadcastMessage extends ChatEventMessage {
+  constructor(message, user, timestamp = null) {
+    super(message, timestamp);
+    this.type = MessageTypes.BROADCAST;
+    this.user = user;
+  }
+
+  buildUserTemplate(chat = null) {
+    // User with an id of -1 is the dummy system user used for server broadcasts
+    if (this.user.id !== -1) {
```
Comment (11k) [inline assets/chat/js/messages/ChatBroadcastMessage.js:14]: Let's make this self-documenting by instead defining a method on `ChatUser` that returns true if the user is the system user. I'd also prefer a guard clause that returns early for the system user to make this method a little easier to read.

**Review by 11k — APPROVED** [review]

File: assets/chat/js/chat.js (hunk 2 above)
Comment (vyneer) [reply assets/chat/js/chat.js]: Done!

File: assets/chat/js/messages/ChatBroadcastMessage.js (hunk 3 above)
Comment (vyneer) [reply assets/chat/js/messages/ChatBroadcastMessage.js]: Done!

File: assets/chat/js/user.js (hunk 1 above)
Comment (vyneer) [reply assets/chat/js/user.js]: Yeah, not necessary - didn't catch it because I had the old user-less messages in history. TY!

**Review by 11k — APPROVED** [review]

**Pull request merged**